ResearchKit/Common/ORKTaskViewController.m

```diff
@@ -981,6 +981,10 @@ - (void)collectToolbarItemsFromViewController:(UIViewController *)viewController
     _pageViewController.toolbarItems = viewController.toolbarItems;
     _pageViewController.navigationItem.leftBarButtonItem = viewController.navigationItem.leftBarButtonItem;
     _pageViewController.navigationItem.rightBarButtonItem = viewController.navigationItem.rightBarButtonItem;
+    if (!_showsProgressInNavigationBar) {
```

Comment:
`progressLabel` determined by two factors:
1. `_showsProgressInNavigationBar`
2. ` [_task respondsToSelector:@selector(progressOfCurrentStep:withResult:)]`

```
Line 910
if (self.showsProgressInNavigationBar &&  [_task respondsToSelector:@selector(progressOfCurrentStep:withResult:)]) {
    ...
}
```

Can you refactor this two factors into a helper method `shouldDisplayProgressLabel` and use it here and `line 910` ?